### PR TITLE
Utility functions to transform enum to string and back

### LINKED ISSUE #102: Utility functions to transform enum to string and back.
Looks like a duplication. I see this code above. It is better to create function which transforms enum to string and back.

_Originally posted by @PokhodenkoSA in https://github.com/IntelPython/dpctl/pull/93#discussion_r499510358_

Example code: https://github.com/IntelPython/dpctl/blob/bb41bb635aa1f3c06c26ac5805a8705bc8f84607/backends/source/dppl_sycl_device_interface.cpp#L62
https://github.com/IntelPython/dpctl/blob/9fcbaa5e97f07053d395f60b3ec7affcc19a3cd1/backends/source/dppl_sycl_platform_interface.cpp#L129
Both examples contain conversion from enum to string. It is a code duplication and could be refactored into single function.
Also it is a good design to create pairs of functions to convert enum to string and string to enum.

### PR DESCRIPTION
**Pull request by 1e-to**
This PR closes #102

### CODE REVIEW

File: backends/source/dppl_sycl_device_interface.cpp [hunk 1]
```diff
@@ -57,26 +58,8 @@ void dump_device_info (const device & Device)
...
-        ss << "custom" << '\n';
-        break;
-    case info::device_type::host:
-        ss << "host" << '\n';
-        break;
-    default:
-        ss << "unknown" << '\n';
-    }
+    auto s = ConvertEnumToStr(devTy);
+    ss << s;
```
Comment (PokhodenkoSA) [inline backends/source/dppl_sycl_device_interface.cpp:62]: Remove `ss`.

Comment (PokhodenkoSA) [reply backends/source/dppl_sycl_device_interface.cpp:62]: Sorry, remove `s`.

File: backends/details/include/dppl_utils_details.h [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+/// This file defines common helper functions used in other places in DPPL.
+//===----------------------------------------------------------------------===//
+
+#pragma once
+
+#include <CL/sycl.hpp>
+using namespace cl::sycl;
+
+std::string ConvertEnumToStr(info::device_type devTy);
+info::device_type ConvertStrToEnum(std::string devTyStr);
```
Comment (PokhodenkoSA) [inline backends/details/include/dppl_utils_details.h:31]: Names of functions is too common. Should be like `device_type2string` and `string2device_type`.

File: backends/source/dppl_sycl_platform_interface.cpp [hunk 3]
```diff
@@ -124,32 +125,14 @@ void DPPLPlatform_DumpInfo ()
...
-                ss << "unknown" << '\n';
-            }
+            ss << ConvertEnumToStr(devTy);
         }
         std::cout << ss.str();
         ++i;
     }
 }
 
+
```
Comment (PokhodenkoSA) [inline backends/source/dppl_sycl_platform_interface.cpp:135]: It is not Python. 1 line between functions is ok.

File: backends/details/source/dppl_utils_details.cpp [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+    } else if(devTyStr == "gpu") {
+        devTy = info::device_type::gpu;
+    } else if(devTyStr == "accelerator") {
+        devTy = info::device_type::accelerator;
+    } else if(devTyStr == "custom") {
+        devTy = info::device_type::custom;
+    } else if(devTyStr == "host") {
+        devTy = info::device_type::host;
+    }
+    return devTy;
```
Comment (PokhodenkoSA) [inline backends/details/source/dppl_utils_details.cpp:71]: If string is not recognized it returns uninitialized variable.

File: backends/details/source/dppl_utils_details.cpp [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+/// This file implements the helper functions defined in dppl_utils_details.h.
+///
+//===----------------------------------------------------------------------===//
+
+#include "dppl_utils_details.h"
+#include <string>
+#include <sstream>
+using namespace cl::sycl;
+
+std::string ConvertEnumToStr(info::device_type devTy)
```
Comment (PokhodenkoSA) [inline backends/details/source/dppl_utils_details.cpp:31]: Please, add docstring.

File: backends/details/source/dppl_utils_details.cpp [hunk 6]
```diff
@@ -0,0 +1,72 @@
...
+///
+/// \file
+/// This file implements the helper functions defined in dppl_utils_details.h.
+///
+//===----------------------------------------------------------------------===//
+
+#include "dppl_utils_details.h"
+#include <string>
+#include <sstream>
+using namespace cl::sycl;
```
Comment (PokhodenkoSA) [inline backends/details/source/dppl_utils_details.cpp:29]:
```suggestion

using namespace cl::sycl;
```

**Comment by PokhodenkoSA** [comment]
Is there a code which uses conversion from string to enum? I do not see is removal in changes.

**Comment by 1e-to** [comment]
> Is there a code which uses conversion from string to enum? I do not see is removal in changes.

**Comment by 1e-to** [comment]
> Is there a code which uses conversion from string to enum? I do not see is removal in changes.

Now it is not uses in project, but Dipto wrote that it is good design to create pairs of functions for conversion.

File: backends/details/include/dppl_utils_details.h (hunk 2 above)
Comment (diptorupd) [reply backends/details/include/dppl_utils_details.h]: How about `DDPL_StrToDeviceType` and `DPPL_DeviceTypeFromStr`? I want to keep the naming convention consistent.

**Review by PokhodenkoSA — APPROVED** [review]
LGTM.

File: backends/details/include/dppl_utils_details.h [hunk 7]
```diff
@@ -0,0 +1,31 @@
...
+/// This file defines common helper functions used in other places in DPPL.
+//===----------------------------------------------------------------------===//
+
+#pragma once
+
+#include <CL/sycl.hpp>
+using namespace cl::sycl;
+
+std::string StrToDeviceType(info::device_type devTy);
+info::device_type DeviceTypeToStr(std::string devTyStr);
```
Comment (PokhodenkoSA) [inline backends/details/include/dppl_utils_details.h:31]: Use `DPPL_` prefix or wrap it in `namespace dppl`. In C we have global namespace for all functions. But this is C++ and internal functions so we could use namspaces.

**Review by PokhodenkoSA — CHANGES_REQUESTED** [review]
Small modifications required.

**Pull request merged**